File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:

    int f(vector<int> &arr,int k){
        int n=arr.size();
        int low=0,high=n-1;
        int first=-1;
        while(low<=high){
            int mid=low+(high-low)/2;
            if (arr[mid]==k){
                first=mid;
                high=mid-1;
            }
            else if (arr[mid]<k){
                low=mid+1;
            }
            else{
                high=mid-1;
            }
        }
        return first;
    }
    int l(vector<int> &arr,int k){
        int n=arr.size();
        int low=0,high=n-1;
        int last=-1;
        while(low<=high){
            int mid=low+(high-low)/2;
            if (arr[mid]==k){
                last=mid;
                low=mid+1;
            }
            else if (arr[mid]< k) low=mid+1;
            else high=mid-1;
        }
        return last;
    }
    vector<int> searchRange(vector<int>& nums, int target) {
        int first=f(nums,target);
        if (first==-1) return {-1,-1};
        int last=l(nums,target);
        return {first,last};
    }
};

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    
    def f(self,arr,x):
        low=0
        high=len(arr)-1
        first=-1
        while(low<=high):
            mid=(low+high)//2
            if(arr[mid]==x):
                first=mid
                high=mid-1
            elif(arr[mid]<x):
                low=mid+1
            else:
                high=mid-1
        return first
    
    def l(self,arr,x):
        low=0
        high=len(arr)-1
        last=-1
        while(low<=high):
            mid=(low+high)//2
            if(arr[mid]==x):
                last=mid
                low=mid+1
            elif(arr[mid]<x):
                low=mid+1
            else:
                high=mid-1
        return last

    def searchRange(self, nums: List[int], target: int) -> List[int]:
        first=self.f(nums,target)
        if(first==-1):
            return [-1,-1]
        last=self.l(nums,target)
        return [first,last]